src/services/messagesService.ts

```diff
@@ -28,39 +31,47 @@ export default class MessagesService {
    * Retrieves all messages for a specific user.
    */
   public async getMessagesByUser(user: User): Promise<Messages> {
-    const messagesPayload = await this.apiClient
+    const response = await this.apiClient
       .getClient(user.fiscal_code)
-      .getMessagesByUser();
+      .getMessagesByUserWithHttpOperationResponse();
 
-    const errorOrApiMessages = GetMessagesByUserOKResponse.decode(
-      messagesPayload
-    );
-    if (isLeft(errorOrApiMessages)) {
-      const errorOrProblemJson = ProblemJson.decode(messagesPayload);
+    const simpleResponse = new SimpleHttpOperationResponse(response);
 
+    if (simpleResponse.status() !== 200) {
```

Comment:
this check is a good candidate to be implemented inside `SimpleHttpOperationResponse` -> `simpleResponse.isStatusOk()` - no need to have a magic number here, is much more readable and also DRY